bndtools.builder/src/org/bndtools/builder/classpath/BndContainerInitializer.java

```diff
@@ -488,7 +488,42 @@ private void addProjectEntry(IPath path, List<IAccessRule> accessRules, List<ICl
 
 		private IPath calculateSourceAttachmentPath(IPath path, File file) {
 			JarInfo info = getJarInfo(file);
-			return info.hasSource ? path : null;
+			IPath sourcePath = info.hasSource ? path : null;
+
+			if (sourcePath == null) {
+				sourcePath = trySourcesJar(file);
+			}
+
+			return sourcePath;
+		}
+
+		/**
+		 * try to see if there is file bsn-sources.jar e.g. as downloaded by the
+		 * Maven repos
+		 *
+		 * @param file
+		 * @return the bsn-sources.jar file if found, otherwise
+		 *         <code>null</code>
+		 */
+		private IPath trySourcesJar(File file) {
+
+			if (file.getName()
+				.endsWith(".jar")) {
```

Comment:
Should maybe also allow for `.JAR`. File names are not case sensitive on all platforms.

Reply:
Good point.

Reply:
I added support for case-insensitive file extension. Although I read that the convention is all lowercase, also enforced by maven central, but unfortunatelly it does not seem to be specified. So I assume there could be uppercase .JAR files out there in enterprise maven repos.